src/app/static/src/tests/integration/projects.itest.ts

```diff
@@ -146,4 +146,25 @@ describe("Projects actions", () => {
             done();
         });
     });
+
+    it("can promote version", async(done) => {
+        const state = initialProjectsState();
+        const commit = jest.fn();
+        const dispatch = jest.fn();
+        
+        await actions.createProject({commit, rootState, state} as any, "v1");
+        expect(commit.mock.calls.length).toBe(2);
+
+        const createdProject = commit.mock.calls[1][0]["payload"];
+        state.currentProject = createdProject;
+        state.currentVersion = createdProject.versions[0];
+        const versionPayload = {version: {projectId: state.currentProject!.id, versionId: state.currentVersion!.id}, name: 'newProject'};
+        
+        await actions.promoteVersion({commit, dispatch, state, rootState} as any, versionPayload);
+        setTimeout(() => {
+            expect(commit.mock.calls.length).toBe(2);
```

Comment:
I still think it would be less confusing to make a fresh commit for calling the action - expect no commits from the action, right?